frontend/tests/artillery/processor.ts

```diff
@@ -0,0 +1,123 @@
+import { readFile } from "fs/promises";
+import { random } from "lodash";
+
+type dataType = {
+  ids: {
+    [key: string]: Array<number>;
+  };
+  queries: Array<string>;
+  pages: Array<string>;
+  status: Array<string>;
+  agencies: Array<string>;
+  funding: Array<string>;
+  eligibility: Array<string>;
+  category: Array<string>;
+};
+type globalVars = {
+  $environment?: string;
+};
+
+type returnVars = {
+  id: number;
+  query: string;
+  route: string;
+  pages: string;
+};
+
+// eslint-disable-next-line @typescript-eslint/require-await
+async function getOppId(context: { vars: dataType & returnVars & globalVars }) {
+  const env = context.vars.$environment as string;
+  context.vars.id =
+    context.vars.ids[env][random(context.vars.ids[env].length - 1)];
+}
+
+// eslint-disable-next-line @typescript-eslint/require-await
+async function get404(context: { vars: returnVars }) {
+  const num = random(10);
+  // ~50% of 404s are opp pages.
+  if (num % 2 !== 0) {
+    context.vars.route = `opportunity/${num}`;
+  } else {
+    context.vars.route = randomString(num);
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/require-await
+async function getStatic(context: { vars: returnVars }) {
+  context.vars.route =
+    context.vars.pages[random(context.vars.pages.length - 1)];
+}
+
+// eslint-disable-next-line @typescript-eslint/require-await
+async function getSearchQuery(context: { vars: returnVars & dataType }) {
+  const { queries, status, agencies, eligibility, category } = context.vars;
+  const qu = `query=${queries[random(queries.length - 1)]}`;
+  const st = `status=${status[random(status.length - 1)]}`;
+  const ag = `agency=${agencies[random(agencies.length - 1)]}`;
+  const cat = `category=${category[random(category.length - 1)]}`;
+  const el = `eligibility=${eligibility[random(eligibility.length - 1)]}`;
+  const pager = `page=${random(5)}`;
+  // Most search params only include the queries, but smaller percent include
+  // filters. This allows configuring that percent for composing the query
+  const weights = [
+    {
+      percent: 50,
+      params: [qu, st, pager],
+    },
+    {
+      percent: 20,
+      params: [qu, st, ag],
+    },
+    {
+      percent: 20,
+      params: [qu, st, ag, cat],
+    },
+    {
+      percent: 10,
+      params: [qu, st, ag, cat, el],
+    },
+  ];
+  // Weight of percents out of 100
+  const hundred = random(100);
+  let total = 0;
+  const selected = weights.find((item) => {
+    total += item.percent;
+    if (hundred <= total) {
+      return true;
+    }
+    return false;
+  });
+  context.vars.query = selected?.params.join("&") as string;
+}
+
+async function loadData(context: { vars: dataType & globalVars }) {
+  // Dev and stage have the same data.
+  const env =
+    context.vars.$environment === "stage" ? "dev" : context.vars.$environment;
+  const envs = new Set(["local", "dev", "prod"]);
```

Comment:
this line too, staging 🙏🏼

Reply:
I'm treating dev and stage as the same since the data is the same. If you use the `stage` flag is uses the `dev` data but still uses the stage routes.

Reply:
That's valid, but I feel like nobody is going to figure that out and then assume that we never load test in staging :/

Reply:
I updated it.